### PR TITLE
Improve quality of code by resolving issues found by quantifiedcode and codacy

### PR DESCRIPTION
**Pull request by CyberDem0n**

### CODE REVIEW

File: patroni/api.py [hunk 1]
```diff
@@ -252,8 +252,8 @@ def get_postgresql_status(self, retry=False):
     def get_tags(self):
         return {'tags': self.server.patroni.tags}
 
-    def log_message(self, format, *args):
-        logger.debug("API thread: " + format % args)
+    def log_message(self, fmt, *args):
```
Comment (CyberDem0n) [inline patroni/api.py]: Codacy Issue found: [Method could be a function](https://www.codacy.com/app/cyberdemn/patroni/file/2033577187/issues/source?bid=3127594&fileBranchId=3129207#l255)

File: patroni/api.py [hunk 2]
```diff
@@ -252,8 +252,8 @@ def get_postgresql_status(self, retry=False):
...
+        logger.debug("API thread: %s - - [%s] %s", self.client_address[0], self.log_date_time_string() + fmt % args)
```
Comment (CyberDem0n) [inline patroni/api.py]: Codacy Issue found: [Not enough arguments for logging format string](https://www.codacy.com/app/cyberdemn/patroni/file/2034615145/issues/source?bid=3127594&fileBranchId=3129207#l256)

File: tests/test_wale_restore.py [hunk 3]
```diff
@@ -92,6 +96,8 @@ def test_run(self):
             with patch.object(self.wale_restore, 'create_replica_with_s3', MagicMock(return_value=0)):
                 self.assertEqual(self.wale_restore.run(), 0)
 
+    @patch('sys.exit', MagicMock())
+    @patch.object(WALERestore, 'run', MagicMock(return_value=0))
     def test_main(self):
-        with patch('sys.exit', MagicMock(return_value=0)):
-            self.assertEqual(main(), None)
+        self.assertEqual(_main(), None)
```
Comment (CyberDem0n) [inline tests/test_wale_restore.py]: Codacy Issue found: [Statement seems to have no effect](https://www.codacy.com/app/cyberdemn/patroni/file/2035720800/issues/source?bid=3127594&fileBranchId=3129207#l103)

File: patroni/ha.py [hunk 4]
```diff
@@ -124,7 +123,9 @@ def follow(self, demote_reason, follow_reason, refresh=True, recovery=False):
         if not self.state_handler.check_recovery_conf(node_to_follow) or recovery:
             self._async_executor.schedule('changing primary_conninfo and restarting')
             self._async_executor.run_async(self.state_handler.follow, (node_to_follow, recovery))
-        return ret
+        if not recovery and self.state_handler.is_leader() or recovery and self.state_handler.role == 'master':
```
Comment (alexeyklyukin) [inline patroni/ha.py]: The state handler may change as a result of self.state_handler.follow.

File: patroni/postgresql.py [hunk 5]
```diff
@@ -394,9 +392,8 @@ def stop(self, mode='fast', block_callbacks=False):
         # patroni.
 
         self.close_connection()
-        if not self.is_running():
-            if not block_callbacks:
-                self.set_state('stopped')
+        if not self.is_running() and not block_callbacks:
+            self.set_state('stopped')
```
Comment (alexeyklyukin) [inline patroni/postgresql.py]: return True should be called always if self.is_running is False.

File: patroni/postgresql.py [hunk 6]
```diff
@@ -556,8 +553,8 @@ def cleanup_archive_status(self):
                         os.unlink(path)
                     elif os.path.isfile(path):
                         os.remove(path)
-                except:
-                    logger.exception("Unable to remove {}".format(path))
+                except OSError:
```
Comment (alexeyklyukin) [inline patroni/postgresql.py]: try for this except should start right before the os.path.isdir call.

**Comment by alexeyklyukin** [comment]
Looks good to me

**Pull request merged**